llm_mining_core/config/server.py

```diff
@@ -33,13 +34,14 @@ def start_llm_server(self):
             "--model", self.model_id,
             "--served-model-name", self.served_model_name,
             "--max-model-len", str(self.MAX_MODEL_LEN),
-            "--chat-template", self.CHAT_TEMPLATE,
             "--uvicorn-log-level", "warning",
             "--disable-log-requests",
             "--dtype", "half",
             "--port", str(self.base_config.port),
             "--tensor-parallel-size",str(self.num_gpus),
-            "--gpu-memory-utilization", self.gpu_memory_util
+            "--gpu-memory-utilization", self.gpu_memory_util,
+            "--tool-call-parser", "hermes",
```

Comment:
what does this param mean? will it change for different models?

Reply:
check this 
```
Automatic Function Calling

To enable this feature, you should set the following flags:

    --enable-auto-tool-choice -- mandatory Auto tool choice. tells vLLM that you want to enable the model to generate its own tool calls when it deems appropriate.
    --tool-call-parser -- select the tool parser to use - currently either hermes or mistral or llama3_json or internlm. Additional tool parsers will continue to be added in the future, and also can register your own tool parsers in the --tool-parser-plugin.
    --tool-parser-plugin -- optional tool parser plugin used to register user defined tool parsers into vllm, the registered tool parser name can be specified in --tool-call-parser.
    --chat-template -- optional for auto tool choice. the path to the chat template which handles tool-role messages and assistant-role messages that contain previously generated tool calls. Hermes, Mistral and Llama models have tool-compatible chat templates in their tokenizer_config.json files, but you can specify a custom template. This argument can be set to tool_use if your model has a tool use-specific chat template configured in the tokenizer_config.json. In this case, it will be used per the transformers specification. More on this [here](https://huggingface.co/docs/transformers/en/chat_templating#why-do-some-models-have-multiple-templates) from HuggingFace; and you can find an example of this in a tokenizer_config.json [here](https://huggingface.co/NousResearch/Hermes-2-Pro-Llama-3-8B/blob/main/tokenizer_config.json)
    ```
https://github.com/vllm-project/vllm/blob/main/docs/source/serving/openai_compatible_server.md#automatic-function-calling